okta/resource_okta_domain.go

```diff
@@ -71,49 +76,63 @@ func resourceDomain() *schema.Resource {
 }
 
 func resourceDomainCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	domain, _, err := getOktaClientFromMetadata(m).Domain.CreateDomain(ctx, buildDomain(d))
+	domainRequest, err := buildDomain(d)
+	if err != nil {
+		return diag.Errorf("failed to build domain: %v", err)
+	}
+	domain, _, err := getOktaV3ClientFromMetadata(m).CustomDomainApi.CreateCustomDomain(ctx).Domain(domainRequest).Execute()
 	if err != nil {
 		return diag.Errorf("failed to create domain: %v", err)
 	}
-	d.SetId(domain.Id)
+	d.SetId(domain.GetId())
+	if brandId, ok := d.GetOk("brand_id"); ok {
+		_, _, err = getOktaV3ClientFromMetadata(m).CustomDomainApi.ReplaceCustomDomain(ctx, d.Id()).UpdateDomain(okta.UpdateDomain{BrandId: brandId.(string)}).Execute()
+		if err != nil {
+			return diag.Errorf("failed to update domain: %v", err)
+		}
+		_, err = getOktaV3ClientFromMetadata(m).CustomizationApi.DeleteBrand(ctx, domain.GetBrandId()).Execute()
```

Comment:
Because whenever a custom domain is created, a brand will be created along with that custom domain. If the customer is using another brand right away, it should be deleted